indexer/osm_editor.cpp

```diff
@@ -227,29 +230,47 @@ void Editor::LoadMapEdits()
       continue;
     }
 
-    for (size_t i = 0; i < sections.size(); ++i)
+    for (auto const & section : sections)
     {
-      for (xml_node node : mwm.child(sections[i].second).children("node"))
+      for (xml_node node : mwm.child(section.second).children("node"))
       {
         try
         {
           XMLFeature const xml(node);
           FeatureID const fid(id, xml.GetOffset());
-          auto & fti = m_features[id][fid.m_index];
-          fti.m_feature = FeatureType::FromXML(xml);
+          FeatureTypeInfo fti;
+
+          if (xml.GetGeomType() != DebugPrint(feature::GEOM_POINT))
+          {
+            // TODO(mgsergio): Check if feature can be read.
+            fti.m_feature = m_featureLoaderFn(fid);
+            fti.m_feature.ApplyPatch(xml);
+          }
+          else
+          {
+            fti.m_feature = FeatureType::FromXML(xml);
+          }
+
           fti.m_feature.SetID(fid);
+
           fti.m_modificationTimestamp = xml.GetModificationTime();
           ASSERT_NOT_EQUAL(my::INVALID_TIME_STAMP, fti.m_modificationTimestamp, ());
           fti.m_uploadAttemptTimestamp = xml.GetUploadTime();
           fti.m_uploadStatus = xml.GetUploadStatus();
           fti.m_uploadError = xml.GetUploadError();
-          fti.m_status = sections[i].first;
+          fti.m_status = section.first;
+
+          /// Install edited feature back to editor.
+          /// This instruction should be after calling m_featureLoaderFn cause
```

Comment:
Call to m_featureLoaderFn tries to load feature by it's ID from the editor's m_features.
That's why insertion into m_features should go AFTER call to m_featureLoaderFn.